File: hw/unda/U3/HW21TemplateMethod/TemplateMethodPy/controller/beverage_test.py
import sys
import os
sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))

from model.tea import Tea
from model.coffee import Coffee

def main():
    print("\nMaking tea ..")
    tea = Tea()
    tea.prepare_recipe()

    print("\nMaking coffee ...")
    coffee = Coffee()
    coffee.prepare_recipe()

if __name__ == "__main__":
    main()

File: hw/jacome/U3/HW21TemplateMethod/JS/view/BeverageTest.js
const Tea = require('../model/Tea');
const Coffee = require('../model/Coffee');
const readline = require('readline');

async function askQuestion(rl, question) {
    return new Promise((resolve) => {
        rl.question(question, (answer) => {
            resolve(answer.toLowerCase());
        });
    });
}

async function main() {
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    try {
        console.log("\nMaking tea...");
        const tea = new Tea();
        tea.prepareRecipe();
        
        const teaAnswer = await askQuestion(rl, "Would you like lemon with your tea (y/n)? ");
        if (teaAnswer === 'y') {
            tea.addCondiments();
        }

        console.log("\nMaking coffee...");
        const coffee = new Coffee();
        coffee.prepareRecipe();
        
        const coffeeAnswer = await askQuestion(rl, "Would you like sugar and milk with your coffee (y/n)? ");
        if (coffeeAnswer === 'y') {
            coffee.addCondiments();
        }
    } finally {
        rl.close();
    }
}

main().catch(console.error);